### DISCUSSION TITLE
How should C# native plugins be integrated

### DISCUSSION THREAD

**Discussion by 527395632** [microsoft/react-native-windows#9981] [Q&A] [answered]
After Auto 0.58, the project was developed from C# language to C++. However, what I am more confused about is if I use C# to develop native plugins, how should I inherit it into ReactNative?

**Comment by chrisglein** [comment, marked as answer]
You can author modules in either C++ or C#. The best way to learn about this is in the [Native Modules documentation](https://microsoft.github.io/react-native-windows/docs/native-modules). There are some considerations for mixing, which you can read about [here](https://microsoft.github.io/react-native-windows/docs/native-code-language-choice).

**Reply by 527395632** [reply]
Thank you very much for @ christilein's reply. I spent two days studying it at the weekend. This problem has been solved. Thank you for your help!
